### PR TITLE
docs: clarify readme and comments

### PR DESCRIPTION
**Pull request by jstirnaman**
docs: clarify READMEs and example code comments:
  - expand comments and descriptions,
  - clarify differences among provided client distributions,
  - explain client distribution used in browser example.

## Checklist

- [x] CHANGELOG.md updated
- [x] Rebased/mergeable
- [x] Commit messages are in [semantic format](https://seesparkbox.com/foundry/semantic_commit_messages)
- [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -41,17 +41,17 @@ InfluxDB 2.0 client consists of two main packages
...
+To write or query InfluxDB, add `@influxdata/influxdb-client` as a dependency to your project using your favorite package manager.
 
 ```
 $ npm install --save @influxdata/influxdb-client
 $ yarn add @influxdata/influxdb-client
 $ pnpm add @influxdata/influxdb-client
 ```
 
-[@influxdata/influxdb-client](./packages/core/README.md) module primarily works in Node.js (main CJS and module ESM), but a browser (browser UMD) distribution is also available therein. If you target browser or [deno](https://deno.land/), use [@influxdata/influxdb-client-browser](./packages/core-browser/README.md).
+[@influxdata/influxdb-client](./packages/core/README.md) module primarily works in Node.js (main CJS and module ESM), but also provides a bundled ESM distribution for browsers. If you target browsers or clients that don't support ESM, see [@influxdata/influxdb-client-browser](./packages/core-browser/README.md) for UMD distributions.
```
Comment (sranka) [inline README.md:52]: I would rather completely revert this change, it is misleading. The way of how libraries are packaged in javascript is complex. The library **code** differs between browser/deno and node.js, Using ESM module from `@influxdata/influxdb-client` would fail because of it. See https://github.com/influxdata/influxdb-client-js/issues/321#issuecomment-812842447 for more details.

**Review by sranka — CHANGES_REQUESTED** [review]
Thank you for this PR, it really improves the docs.

File: examples/env_browser.js [hunk 2]
```diff
@@ -1,16 +1,23 @@
 // eslint-disable-next-line no-undef
+/** 
+ * The following configuration is used in the browser examples
+ * (index.html and giraffe.html).
+ *
+ * Replace the values with your own InfluxDB values.
+ */
```
Comment (sranka) [inline examples/env_browser.js:7]: This change is required to appease linter, the CI tests also control the style and format of examples.

```suggestion
/*
 * The following configuration is used in the browser examples
 * (index.html and giraffe.html).
 *
 * Replace the values with your own InfluxDB values.
 */
// eslint-disable-next-line no-undef
```

**Review by sranka — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (jstirnaman) [reply README.md:52]: @sranka, I wrote it like this because `@influxdata/influxdb-client` is used in https://github.com/influxdata/influxdb-client-js/blob/master/examples/index.html, which I found suspicious, but seems to work for the example.

Comment (sranka) [reply README.md:52]: > @sranka, It is not misleading unless `@influxdata/influxdb-client` doesn't support it (it seems to support it):
> 
> a) https://github.com/influxdata/influxdb-client-js/blob/master/examples/index.html loads `@influxdata/influxdb-client` ESM with `<script type="module">`. The example app seems to work as expected, though I did find it suspicious, or at least confusing, that example/index.html doesn't use `@influxdata/influxdb-client-browser`.
> 
the examples load a browser module from URL `https://unpkg.com/@influxdata/influxdb-client/dist/index.browser.mjs` , it is the same as `https://unpkg.com/@influxdata/influxdb-client/dist/index.browser.mjs` (and might not be that confusing).  ... `@influxdata/influxdb-client-browser` is especially useful for in bundlers like webpack or rollup that read `package.json` to pick up the preferred distribution based on its module/main/browser properties.
> b) ES module support with `<script type="module">` is [becoming standard](https://caniuse.com/es6-module) across modern browsers (as in Node.js). It's also commonly used in Webpack builds.
> 
sure, still node.js and browser distributions are different files
> c) ES modules are the [standard in Deno](https://deno.land/manual#comparison-to-nodejs).

sure, and typescript modules

Comment (jstirnaman) [reply README.md:52]: > > @sranka, It is not misleading unless `@influxdata/influxdb-client` doesn't support it (it seems to support it):
> > a) https://github.com/influxdata/influxdb-client-js/blob/master/examples/index.html loads `@influxdata/influxdb-client` ESM with `<script type="module">`. The example app seems to work as expected, though I did find it suspicious, or at least confusing, that example/index.html doesn't use `@influxdata/influxdb-client-browser`.
> 
> the example loads a browser module from URL `https://unpkg.com/@influxdata/influxdb-client/dist/index.browser.mjs`, it is the same as `https://unpkg.com/@influxdata/influxdb-client-browser/dist/index.browser.mjs` (which might not be that confusing). ... `@influxdata/influxdb-client-browser` is especially useful in bundlers like webpack or rollup that read `package.json`'s module/main/browser properties to pick up the preferred distribution.
> 
> > b) ES module support with `<script type="module">` is [becoming standard](https://caniuse.com/es6-module) across modern browsers (as in Node.js). It's also commonly used in Webpack builds.
> 
> sure, still node.js and browser modules are different files
> 
> > c) ES modules are the [standard in Deno](https://deno.land/manual#comparison-to-nodejs).
> 
> sure, and typescript modules

Thanks for the clarification. I'll edit the PR. Still, I would argue that recommending the use of `@influxdata/influxdb-client-browser` and then not using it in example/index.html is confusing, whether or not it's the same distribution.

I would also suggest removing or clarifying this:
>>, but a browser (browser UMD) distribution is also available therein

and only encourage `@influxdata/influxdb-client-browser` for browsers.

Comment (sranka) [reply README.md:52]: > Thanks for the clarification. I'll edit the PR. Still, I would argue that recommending the use of `@influxdata/influxdb-client-browser` and then not using it in example/index.html is confusing, whether or not it's the same distribution.
> 
Yes, it is a nice improvement.

> I would also suggest removing or clarifying this:
> 
> > > , but a browser (browser UMD) distribution is also available therein
> 
> and only encourage `@influxdata/influxdb-client-browser` for browsers.
It was originally created this way to support isomorphic JavaScript apps. Sure, such a technical detail can be skipped for the sake of clarity/simplicity.

File: README.md [hunk 3]
```diff
@@ -49,7 +49,11 @@ $ yarn add @influxdata/influxdb-client
 $ pnpm add @influxdata/influxdb-client
 ```
 
-[@influxdata/influxdb-client](./packages/core/README.md) module primarily works in Node.js (main CJS and module ESM), but also provides a bundled ESM distribution for browsers. If you target browsers or clients that don't support ESM, see [@influxdata/influxdb-client-browser](./packages/core-browser/README.md) for UMD distributions.
+If you target Node.js, use [@influxdata/influxdb-client](./packages/core/README.md).
```
Comment (jstirnaman) [inline README.md:52]: @sranka Please review this change. Thanks!

**Review by sranka — APPROVED** [review]
Nice

**Comment by jstirnaman** [comment]
Thanks!
